src/functions/module_logical.ts

```diff
@@ -101,27 +107,23 @@ export const IFERROR = {
     ),
   ],
   returns: ["ANY"],
-  computeFormat: (
+  computeValueAndFormat: function (
     value: () => PrimitiveArg,
     valueIfError: () => PrimitiveArg = () => ({ value: "" })
-  ) => {
-    try {
-      return value()?.format;
-    } catch (e) {
-      return valueIfError()?.format;
-    }
-  },
-  compute: function (
-    value: () => PrimitiveArgValue,
-    valueIfError: () => PrimitiveArgValue = () => ""
-  ): FunctionReturnValue {
+  ): FunctionReturn {
     let result;
     try {
       result = value();
     } catch (e) {
       result = valueIfError();
     }
-    return result === null || result === undefined ? "" : result;
+    if (result === undefined) {
+      return { value: "" };
+    }
+    if (result.value === null) {
+      result.value = "";
```

Comment:
not tested

Reply:
Here ?: https://github.com/odoo/o-spreadsheet/blob/master-compute-format-and-value-round-2/tests/functions/module_logical.test.ts#L123